src/packet_macros.rs

```diff
@@ -219,3 +237,18 @@ macro_rules! write_packet_field {
         $stream.write_chunk_section($value)
     };
 }
+
+macro_rules! translate_packet_field {
+    ($value:expr, $transdata:expr, Untranslated) => {
+        $value
+    };
+    ($value:expr, $transdata:expr, EntityId) => {
+        $value + ($transdata.entity_id_block * ENTITY_ID_BLOCK_SIZE)
```

Comment:
this might be an issue if we every have two fields that are translated the same way but have a different underlying type- we'll have to figure that out once we get to that point, which I'm pretty sure we will. Hopefully throwing a few 'into()'s will do the trick

Reply:
Also we probably shouldn't be jamming actual translation logic into the macro class, these should just call methods, but since our translations are so simple and few in number this will be fine for now. Once this gets unwieldy we'll refactor